charms/istio-pilot/src/charm.py

```diff
@@ -124,13 +125,19 @@ def __init__(self, *args):
         # ---- WARNING: this feature is not recommended, but is supported in 1.17-1.21.
         # ---- For details please refer to canonical/istio-operators#380.
         # ---- FIXME: Remove this block after releasing 1.21.
-        # Save SSL information and reconcile
-        self.framework.observe(self.on.set_tls_action, self.set_tls)
+        # Save TLS information and reconcile
+        self._tls_secret_id = self.config.get("tls-secret-id")
         self.framework.observe(self.on.secret_changed, self.reconcile)
 
-        # Remove SSL information and reconcile
-        self.framework.observe(self.on.unset_tls_action, self.unset_tls)
-        self.framework.observe(self.on.secret_remove, self.reconcile)
+        # Remove TLS information and reconcile
+        # NOTE: contrary to what users would think, removing a user secret
+        # via the CLI "juju remove-secret <secret:secret-ID> will not trigger
+        # a secret_remove event.
+        # The way I have found a workaround is to remove the config option
+        # that stores the secret ID, that way the logic for handling a config-changed
+        # runs and updates the bits where the secret contents are used, in our case
+        # the TLS configuration of the Gateway.
+        # self.framework.observe(self.on.secret_remove, self.reconcile)
```

Comment:
I wonder about keeping `on.secret_remove` here even thought it doesn't work as we expect atm.  I could see the current behaviour being a bug and it being fixed in a future juju version

If we do `.observe(secret_remove)`, I can't think of what we might break by mistake.  So I think its low risk to uncomment this.  wdyt?
```suggestion
        self.framework.observe(self.on.secret_remove, self.reconcile)
```

Reply:
Let's see what juju team says, otherwise I 'm fine with both approaches.

Reply:
Removing it as the docs description is entirely different from the way we expect (this event to be triggered when the secret is removed from the model). If this behaviour is changed in the future, I can add it.